java/arcs/core/entity/EntityDereferencerFactory.kt

```diff
@@ -94,4 +94,10 @@ class ForeignEntityDereferencer(
     }
     return null
   }
+
+  override fun equals(other: Any?) = when(other) {
```

Comment:
Whenever you override equals you need to override hashCode too, otherwise weird things happen.

https://stackoverflow.com/a/2265637

Reply:
Woa, that's actually kinda cool. I overwrote the hashCode function